### PR TITLE
Fix for STRICT warnings in PHP 5.4

### PR DESCRIPTION
**Pull request by cuppett**
Resubmitting this for a minimal diff instead of breaking all the line endings.

Received several STRICT warnings using PHP 5.4 with this class.  I tried to keep the code logic mostly the exact same and just get the signature errors to go away.

### CODE REVIEW

**Comment by faizshukri** [comment]
file Model\Behavior\SchemalessBehavior.php also have to be fixed..

**Comment by cuppett** [comment]
Yes.  I can submit a separate pull request for that.  I didn't particularly
notice it with my usage, but I can analyze the interface used.

**Comment by cuppett** [comment]
Okay, these changes complete it.

File: Model/Datasource/MongodbSource.php [hunk 1]
```diff
@@ -414,12 +414,20 @@ public function begin() {
...
  * @return array
  * @access public
  */
-	public function calculate(&$Model) {
-		return array('count' => true);
+	public function calculate(Model $model, $func, $params = array()) {
+	
+		switch (strtolower($func)) {
+			case 'count':
+				array('count' => true);
```
Comment (moderndeveloperllc) [inline Model/Datasource/MongodbSource.php]: Did you forget your return statement here?

Comment (cuppett) [inline Model/Datasource/MongodbSource.php]: Looks like it.  I'll clean it up.

**Comment by cuppett** [comment]
Okay, one clean commit 9d1ffb56f220fd5ae2af0f70c4f03b179bb11cfd to pull.

**Comment by k1LoW** [comment]
Merge #65

**Pull request closed without merging**